### PR TITLE
add better failure handling for state promises in authoring index

### LINKED ISSUE #781: delete in authoring can create an issue when search-index is rebuilt
Steps to reproduce:
1. Have tweek environment with lots of keys so index will be slow.
2. Do some editing
3. Delete item from editor (after archive)

There's a chance you'll experience crash in the authoring in it'll be stuck in a bad state - ManifestPromise/IndexPromise can be stuck in a rejected state.
https://github.com/Soluto/tweek/blob/master/services/authoring/src/search-index/index.ts

Fix: need to check how the indexer works

### PR DESCRIPTION
**Pull request by Yshayy**
add failure handling for index state promises (use the previous version until next save)

Partly fix #781

### CODE REVIEW

File: services/authoring/src/search-index/index.ts [hunk 1]
```diff
@@ -72,8 +72,8 @@ export default {
     return dependentsPromise.then(({ aliasIndex, dependencyIndex }) => ({ usedBy: dependencyIndex[key] || [], aliases: aliasIndex[key] || [] }));
   },
   refreshIndex: (repoDir) => {
-    indexPromise = refreshIndex(repoDir);
-    manifestPromise = getManifests(repoDir);
+    indexPromise = refreshIndex(repoDir).catch((x=>()=>x)(indexPromise));
```
Comment (nataly87s) [inline services/authoring/src/search-index/index.ts]: why not simply `.catch(() => indexPromise)`?

Comment (Yshayy) [reply services/authoring/src/search-index/index.ts]: It'll be a bit ugly because i'll need to define lots of currentBlaPromise.

Comment (nataly87s) [reply services/authoring/src/search-index/index.ts]: then maybe 
```js
const persistPromise = x => () => x;
indexPromise = refreshIndex(repoDir).catch(persistPromise(indexPromise));
```

Comment (Yshayy) [reply services/authoring/src/search-index/index.ts]: Maybe:```makeRef(indexPromise)?```

**Pull request merged**